app/controlplane/internal/service/project.go

```diff
@@ -302,6 +302,67 @@ func (s *ProjectService) RemoveMember(ctx context.Context, req *pb.ProjectServic
 	return &pb.ProjectServiceRemoveMemberResponse{}, nil
 }
 
+// UpdateMemberRole updates the role of a user or group in a project.
+func (s *ProjectService) UpdateMemberRole(ctx context.Context, req *pb.ProjectServiceUpdateMemberRoleRequest) (*pb.ProjectServiceUpdateMemberRoleResponse, error) {
+	currentOrg, err := requireCurrentOrg(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Make sure the provided project exists and the user has permission to update member roles in it
+	_, err = s.userHasPermissionOnProject(ctx, currentOrg.ID, req.GetProjectReference(), authz.PolicyProjectAddMemberships)
```

Comment:
I am assuming that `PolicyProjectAddMemberships` is the only one we have, not an update membership policy?

Reply:
We also have `ProjectMemberRoleUpdatedType`